src/psyclone/psyir/transformations/acc_update_trans.py

```diff
@@ -162,12 +161,23 @@ def _add_updates_to_schedule(self, sched):
         for child in sched[:]:
             if isinstance(child, self._ACC_IGNORE):
                 continue
-            if not child.walk(self._ACC_COMPUTE + self._MAY_COMPUTE):
+            # Avoid matching IntrinsicCall unless it is allocate or deallocate.
+            found = False
+            for node in child.walk(Node):
+                if isinstance(node, self._ACC_COMPUTE) or \
+                       type(node) in self._MAY_COMPUTE or \
+                       (isinstance(node, IntrinsicCall) and
+                        node.routine.name in ["ALLOCATE", "DEALLOCATE"]):
+                    found = True
+                    break
+            if not found:
```

Comment:
Yes, I forgot to tidy this up. Implemented a static method as suggested. What I have done in this PR is replicate the existing functionality, which is that allocate and deallocate are treated in the same way as calls. I agree that this might not be what is wanted in the future but I would argue that is for another issue/pr - particularly as Andy and Nuno were not sure how to treat them in this transformation when I asked. I'm not sure adding an example makes sense as I think the logic will be changed in the future issue/pr mentioned earlier.